allennlp/data/dataset_readers/stanford_sentiment_tree_bank.py

```diff
@@ -0,0 +1,136 @@
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import TokenIndexer, SingleIdTokenIndexer
+from allennlp.data.tokenizers import Token
+from allennlp.common.checks import ConfigurationError
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("sst_tokens")
+class StanfordSentimentTreeBankTokensDatasetReader(DatasetReader):
+    """
+    Reads tokens and their sentiment labels from the Stanford Sentiment Treebank.
+
+    Parameters
+    ----------
+    token_indexers : ``Dict[str, TokenIndexer]``, optional (default=``{"tokens": SingleIdTokenIndexer()}``)
+        We use this to define the input representation for the text.  See :class:`TokenIndexer`.
+        Note that the `output` tags will always correspond to single token IDs based on how they
+        are pre-tokenised in the data file.
+    use_subtrees : ``bool``, optional, (default = ``False``)
+        Whether or not to use sentiment-tagged subtrees.
+    granularity : ``str``, optional (default = ``"5-class"``)
+        One of ``"5-class"``, ``"3-class"``, or ``"2-class"``, indicating the number
+        of sentiment labels to use. The Stanford Sentiment Treebank comes with labels
+        from 0 to 4. ``"5-class"`` uses these labels as is. ``"3-class"`` converts the
+        problem into one of identifying whether a sentence is negative, positive, or
+        neutral sentiment. In this case, 0 and 1 are grouped as label 0 (negative sentiment),
+        2 is converted to label 1 (neutral sentiment) and 3 and 4 are grouped as label 2
+        (positive sentiment). ``"2-class"`` turns it into a binary classification problem
+        between positive and negative sentiment. 0 and 1 are grouped as the label 0
+        (negative sentiment), 2 (neutral) is discarded, and 3 and 4 are grouped as the label 1
+        (positive sentiment).
+    lazy : ``bool``, optional, (default = ``False``)
+        Whether or not instances can be consumed lazily.
+    """
+    def __init__(self,
+                 token_indexers: Dict[str, TokenIndexer] = None,
+                 use_subtrees: bool = False,
+                 granularity: str = "5-class",
+                 lazy: bool = False) -> None:
+        super().__init__(lazy=lazy)
+        self._token_indexers = token_indexers or {'tokens': SingleIdTokenIndexer()}
+        self._use_subtrees = use_subtrees
+        allowed_granularities = ["5-class", "3-class", "2-class"]
+        if granularity not in allowed_granularities:
+            raise ConfigurationError("granularity is {}, but expected one of: {}".format(
+                    granularity, allowed_granularities))
+        self._granularity = granularity
+
+    @overrides
+    def _read(self, file_path):
+        with open(cached_path(file_path), "r") as data_file:
+            logger.info("Reading instances from lines in file at: %s", file_path)
+            for line in tqdm.tqdm(data_file.readlines()):
```

Comment:
Having `tqdm` in here does not play nicely with lazy datasets.  We've removed this from all of our concrete dataset readers, and put it in the base class.  Also, just FYI, we've also added a global configuration of `tqdm`, so when you _do_ want to use `tqdm`, you should do it through our `Tqdm.tqdm()` interface, instead of using `tqdm.tqdm` directly.